### PR TITLE
Setup for Firebase Test Lab.

### PR DESCRIPTION
**Pull request by tikurahul**
* Setup an FTL Workflow.
* Setup Cloud Functions that handle callbacks from FTL.
* Expose a REST API to re-run processing of Macro Benchmark results
  in case of flakes & for debugging.

### CODE REVIEW

File: .github/workflows/firebase_test_lab.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+name: Runs Macro Benchmarks on Firebase Test Lab
```
Comment (ChrisCraik) [inline .github/workflows/firebase_test_lab.yml:1]: Macrobenchmarks

File: MacrobenchmarkSample/functions/src/handlers.ts [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+import { MetricServiceClient } from "@google-cloud/monitoring";
+import { uploadMetrics } from "./monitoring";
+import * as express from 'express';
+
+// The key under which well known device configurations are saved.
+// This is an environment variable.
+const DEVICE_CONFIGURATIONS = 'DEVICE_CONFIGURATIONS';
+
+// The name of the output JSON file.
+const BENCHMARK_RESULT_FILE = 'androidx.benchmark.macrobenchmarks.test-benchmarkData.json';
```
Comment (ChrisCraik) [inline MacrobenchmarkSample/functions/src/handlers.ts:16]: should package be a parameter? how do you specify custom parameters to the functions?

File: MacrobenchmarkSample/functions/src/schema.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
+/**
+ * The Macro Benchmark Context.
```
Comment (ChrisCraik) [inline MacrobenchmarkSample/functions/src/schema.ts:2]: Macrobenchmark, here and elsewhere

File: MacrobenchmarkSample/functions/src/schema.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
+/**
```
Comment (ChrisCraik) [inline MacrobenchmarkSample/functions/src/schema.ts:1]: Technically this schema isn't macrobenchmark specific, except for the specific metric names.

suggest referring to this as general benchmark schema, but make a note in the `'metrics':{}` definition that it only supports macrobenchmark metrics currently.

File: MacrobenchmarkSample/functions/src/handlers.ts (hunk 2 above)
Comment (tikurahul) [reply MacrobenchmarkSample/functions/src/handlers.ts]: Made this an environment variable.

File: .github/workflows/firebase_test_lab.yml (hunk 1 above)
Comment (tikurahul) [reply .github/workflows/firebase_test_lab.yml:1]: Done

File: MacrobenchmarkSample/functions/src/schema.ts (hunk 4 above)
Comment (tikurahul) [reply MacrobenchmarkSample/functions/src/schema.ts:1]: We can make this more flexible. This is not intended to be a general purpose schema definition.

File: MacrobenchmarkSample/functions/src/schema.ts (hunk 3 above)
Comment (tikurahul) [reply MacrobenchmarkSample/functions/src/schema.ts]: Done

**Pull request merged**